test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Execution/UnitTestRunnerTests.cs

```diff
@@ -412,6 +412,10 @@ public void TestMethod()
             }
         }
 
+        private class DummyTestClassAttribute : UTF.TestClassAttribute
```

Comment:
Please add [DummyTestClass] to class DummyTestClassWithCleanupMethods(Line 388) as well.

Reply:
That is weird.

Reply:
Indeed it is. I did even try debugging the unit test that was using the `DummyTestClassWithCleanupMethods` with the `[DummyTestClass]` and `[UTF.TestClass]` but could not observe nothing. and got same behavior except for the part of the assemblyCleanup thing. Notice that when invoking the ClassCleanup it works properly.

Reply:
Tried debugging this. Basically, when we get [DeclaredMethods](https://github.com/Microsoft/testfx/blob/master/src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs#L410) for `DummyTestClassWithCleanupMethods`, it is not returning AssemblyCleanup() method in the list of methods for that type and hence AssemblyCleanup() is not getting called. Now, the question is why AssemblyCleanup() is not present in the list when we mention `[DummyTestClass]` and is present in case of [UTF.TestClass]

Reply:
That's the same question I asked to myself...
I created a mstest project, added a class `CustomTestClass` that inherits from `TestClass`, and set up the assembly/class initalize and cleanup methods. It looks like there could be an issue when a test class is inheriting from a custom attribute that inherits from TestClassAttribute.
Have you seem any issues regarding to that?

Here's the debug output with `CustomTestClass` attribute:
`expectedValue` is a private static int field declared in the test class

> ClassInitialize: before -> 0
> ClassInitialize: after -> 321
> TestMethod: expectedValue -> 321
> 
> Assert.AreEqual failed. Expected:<123>. Actual:<321>. 
>  
> ClassCleanup: expectedValue -> 321

_**notice** that with the custom attribute, neither the assembly initialize nor the cleanup were executed_

Here's the debug output with `TestClass` attribute:

> AssemblyInitialize: before -> 0
> AssemblyInitialize: after -> 123
> ClassInitialize: before -> 123
> ClassInitialize: after -> 123
> TestMethod: expectedValue -> 123
> ClassCleanup: expectedValue -> 123
> AssemblyCleanup: expectedValue -> 123

Reply:
@jayaranigarg any updates?
I'm thinking of opening an issue with my inputs in above comments? Would that be alright?

Reply:
@parrainc : That should be fine. Please raise a separate issue for this and let's get this PR in.

Reply:
@parrainc : That should be fine. Please raise a separate issue for this and let's get this PR in.

Reply:
@parrainc : That should be fine. Please raise a separate issue for this and let's get this PR in.

Reply:
@parrainc : That should be fine. Please raise a separate issue for this and let's get this PR in.

Reply:
@parrainc : That should be fine. Please raise a separate issue for this.

Reply:
test